Update Router readme (#1071)
Shopify/quilt: packages/react-router/README.md

### `<Router />`

> Rendering the `Router` component at the top-level of your application will create the router that is provided to the rest of the React tree. It takes a single prop, `location`, that represents the current location to be used in the initial server-side render of the application.

Reviewer: ```suggestion
Rendering the `Router` component at the top-level of your application will create the router that is provided to the rest of the React tree. It takes an optional single prop, `location`, that can be left `undefined` during client side rendering, but need to be passed in to represents the current location in the initial server-side render of the application.
```

> ```

Reviewer: ```suggestion
```tsx
```

import React from 'react';
import {Router} from '@shopify/react-router';

> export function App({location}) {

Reviewer: ```suggestion
export function App({location}: {location?: string}) {
```
